core/arch/arm/plat-nuvoton/utils/npcm845x_utils.c

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022-2023, ARM Limited and Contributors. All rights reserved.
+ * Copyright (C) 2022-2023 Nuvoton Ltd.
+ */
+
+#include <kernel/linker.h>
+#include <utils/npcm845x_trace.h>
+#include <npcm845x_utils.h>
+
+void print_version(void)
+{
+	COLOR_MSG(COLOR_MAGENTA);
+	COLOR_MSG(">================================================");
+	COLOR_MSG("OP-TEE OS Version %s", core_v_str);
+	COLOR_MSG(">================================================");
+	COLOR_MSG(COLOR_NORMAL);
```

Comment:
What's wrong with the existing macros: `IMSG()` or `IMSG_RAW()` or `MSG_RAW()`?

Reply:
Have you tried the macros I suggested? I see no special handling for color escape codes in the new functions you have added (`trace_color_printf()` and `trace_color_vprintf()`).

Reply:
I can propose a new `CFG_CONSOLE_WITH_COLOR` when enabled, `IMSG()` can be magenta, `EMSG()` be red, `DMSG()` be ... . We disabled, no colorization of the console messages.

Reply:
```diff
diff --git a/core/arch/arm/kernel/boot.c b/core/arch/arm/kernel/boot.c
index 82a57ff33..14a255a56 100644
--- a/core/arch/arm/kernel/boot.c
+++ b/core/arch/arm/kernel/boot.c
@@ -1433,6 +1433,9 @@ static bool cpu_nmfi_enabled(void)
 #endif
 }
 
+#define COLOR_NORMAL	"\x1B[0m"
+#define COLOR_MAGENTA	"\x1B[35m"
+
 /*
  * Note: this function is weak just to make it possible to exclude it from
  * the unpaged area.
@@ -1451,7 +1454,9 @@ void __weak boot_init_primary_late(unsigned long fdt,
 	update_external_dt();
 	configure_console_from_dt();
 
+	IMSG_RAW(COLOR_MAGENTA);
 	IMSG("OP-TEE version: %s", core_v_str);
+	IMSG_RAW(COLOR_NORMAL);
 	if (IS_ENABLED(CFG_WARN_INSECURE)) {
 		IMSG("WARNING: This OP-TEE configuration might be insecure!");
 		IMSG("WARNING: Please check https://optee.readthedocs.io/en/latest/architecture/porting_guidelines.html");
```

![image](https://github.com/OP-TEE/optee_os/assets/8288195/976745f8-7cd3-44d6-96f7-112a21a9be26)

Reply:
Thank you, MSG(COLOR_MAGENTA) and after that MSG(COLOR_NORMAL) worked well, so I remove the added macros and functions. If you wish to transfer all color codes to kernel, no problem at all.

Reply:
I would rather have the colorization characters integrated straight from the inside `xMSG()` macros.

Reply:
I'm sorry, but do we really need to make the log look like a christmas tree? To be honest I find colors annoying and it makes it harder to read. We also have to consider that everyone can't distinguish all colors.

Reply:
> I'm sorry, but do we really need to make the log look like a christmas tree? To be honest I find colors annoying and it makes it harder to read. We also have to consider that everyone can't distinguish all colors.

Fair enough, IMO this color thing should be left out of this PR and dealt with as Etienne proposed, with an optional CFG.

Reply:
Isn't that what `E/` is supposed to be used for?
With these color codes, I expect that we'll find ugly escape sequences in the log if feeding it into `less` or such.

Reply:
I'd rather have the color part of this PR in a separate PR.

Reply:
> With these color codes, I expect that we'll find ugly escape sequences in the log if feeding it into less or such

I agree. A `CFG_` config switch default disabled would be far better than enforcing such colorization.
@rutigl, I think it can be a good excercie for you to propose a generic P-R for that purpose.